### ISSUE TITLE
Better support inventory group of groups and group / host variables 

### ISSUE THREAD

**Issue by claudio-viola** [metacloud/molecule#1011]
# Issue Type
- Feature request

#  Current Behaviour
We have to define groups and children in molecule.yml

   platforms:
    - name: platform1
    groups:
     - group1
     - group2
     - group3
   children:
     - group4
Then we have to create 
group_vars/Group4
group_vars/Group1
host_vars/platform1

# Desired Behaviour
Be able to import ansible group of group / children variable configuration from a yaml file in the molecule.yml configuration
platforms:
  - name: platform1
     inventory_config_import: ../

Basically add the ability to import the configuration from a yaml file structured like this https://docs.ansible.com/ansible/latest/intro_inventory.html#groups-of-groups-and-group-variables

[atlanta]
host1
host2

[raleigh]
host2
host3

[southeast:children]
atlanta
raleigh

[southeast:vars]
some_server=foo.southeast.example.com
halon_system_timeout=30
self_destruct_countdown=60
escape_pods=2

[usa:children]
southeast 
northeast
southwest
northwest

**Comment by retr0h** [comment]
I believe importing from a yaml file is an Ansible 2.4 only feature.  For now we are holding off on such a feature.

You can also supply the child data to children instead of creating files on disk.

```
platforms:
  - name: host1
    image: centos:latest
    groups:
      - atlanta
    children:
      - southeast
      - usa
  - name: host2
    image: centos:latest
    groups:
      - atlanta
      - raleigh
    children:
      - southeast
      - usa
  - name: host3
    image: centos:latest
    groups:
      - raleigh
    children:
      - southeast
      - usa
provisioner:
  name: ansible
  inventory:
    group_vars:
      southeast:
        some_server: foo.southeast.example.com
        halon_system_timeout: 30
        self_destruct_countdown: 60
        escape_pods: 2
```

An inventory file is created as such:

```
cat molecule/default/.molecule/ansible_inventory.yml
# Molecule managed

---
all:
  hosts:
    host1: &id001
      ansible_connection: docker
    host2: &id003
      ansible_connection: docker
    host3: &id004
      ansible_connection: docker
atlanta:
  children:
    southeast:
      hosts:
        host1: *id001
        host2: &id002
          ansible_connection: docker
    usa:
      hosts:
        host1: *id001
        host2: *id002
  hosts:
    host1: *id001
    host2: *id002
raleigh:
  children:
    southeast:
      hosts:
        host2: *id003
        host3: *id004
    usa:
      hosts:
        host2: *id003
        host3: *id004
  hosts:
    host2: *id003
    host3: *id004
ungrouped:
  vars: {}
```

And group vars are created automatically.

```
$ tree molecule/default/.molecule/group_vars
molecule/default/.molecule/group_vars
└── southeast

0 directories, 1 file
$ cat molecule/default/.molecule/group_vars/southeast
# Molecule managed

---
escape_pods: 2
halon_system_timeout: 30
self_destruct_countdown: 60
some_server: foo.southeast.example.com
```

**Comment by claudio-viola** [comment]
Thanks for the answer I am using ansible 2.2.3.0 and those inventories are loaded correctly!
Thanks for the tip although I prefer files actually because that keeps the life of the developer easier instead of merging those with molecule.

**Comment by retr0h** [comment]
Down the road we may add such a feature.

**Comment by retr0h** [comment]
You can always use [include_vars](http://docs.ansible.com/ansible/latest/include_vars_module.html) if you wish to maintain and load a yaml.

**Comment by claudio-viola** [comment]
thank you so much , great job. can be closed for me! 👍
